lib/manageiq/cross_repo/runner.rb

```diff
@@ -58,13 +64,17 @@ def system!(*args)
         repo = Dir.pwd.split("/").last(2).join("/")
         puts "\e[36mDEBUG: #{repo} - #{args.join(" ")}\e[0m"
       end
-      exit($?.exitstatus) unless system(*args)
+
+      Process.wait(spawn(*args))
```

Comment:
I ended up going with spawn because I could control the input/output pipes, specifically I could set the output pipe to `$stdout` so that output was logged as it happened not after the command exited.

We could enhance AwesomeSpawn to do this but it explicitly filters out `:in`, `:out`, and `:err`.

Reply:
> We could enhance AwesomeSpawn to do this but it explicitly filters out `:in`, `:out`, and `:err`.

Fun fact:  I tried doing just that to an extent already

https://github.com/ManageIQ/awesome_spawn/pull/41

But ended up not being a priority nor what I ended up using to support the backups in the end (actually did the file splitting in ruby and worked with FIFOs).

That said, based on how you took my suggestion here and actually made it into something cool, maybe you will have a better go at it 😄 .

Reply:
That is interesting @NickLaMuro I really like the nested hashes for pipes
I was thinking about adding something like `:in_pipe` as another option to `:in_data`, and also `:out_pipe` and `:err_pipe`

Reply:
`Open3` does all of that already, which is why it's kind of not needed for awesome_spawn.  Or at least, that's been my hesitation with adding more there.

Reply:
> `Open3` does all of that already...

100% with this, and I am actually fine with that being a reason not to merge.

However, generally speaking, when we ("the MIQ core Developers") suggest what to use for shell out execution when code reviewing, we standardize (generally) around `AwesomeSpawn`.  That said, the nuanced message of "this doesn't have all of the features of `spawn`/`Open3`/etc." is usually lost (in my opinion) to others since it isn't obvious to other "non-fulltime-rubyists" other options exist.

* * *

`AwesomeSpawn` shines at command sanitization, and has a nicer Ruby interface for passing CLI args, but lacks on IO when compared to `Kernel.spawn` and `Open3`.  Possibly that is something we should add to the `README.md` (of that project) just to make it more clear of it's use case.  Thoughts?